### DISCUSSION TITLE
When password contains @, it cannot resolve the host

### DISCUSSION THREAD

**Discussion by Irvingouj** [PostgREST/postgrest#2659] [Q&A] [answered]
### Environment

* PostgreSQL version: postgres:14.1-alpine
* PostgREST version: 10.1.2
* Operating system: ubuntu 20.04 LTS

### Description of issue
when password contains @, it does not resolve the hostname correctly
```
10/Feb/2023:15:32:30 -0500: Attempting to connect to the database...
10/Feb/2023:15:32:35 -0500: {"code":"PGRST000","details":"could not translate host name \"Ojy233@localhost\" to address: Try again\n","hint":null,"message":"Database connection error. Retrying the connection."}
```

(Expected behavior vs actual behavior)
expected: connect to the database
actual: shown above
(Steps to reproduce: Include a minimal SQL definition plus how you make the request to PostgREST and the response body)
create a user with a password containing @, for example 123456@postgres

**Comment by steve-chavez** [comment, marked as answer]
You need to percent encode `@` as `%40` when using it on the PostgreSQL connection string. See the [pg docs](https://www.postgresql.org/docs/12/libpq-connect.html#LIBPQ-CONNSTRING).

Previously discussed on:

- https://github.com/PostgREST/postgrest/issues/1244
- https://github.com/PostgREST/postgrest/issues/1646
